### PR TITLE
Event Horizon Anti-Existential Beam Rifle. The ultimate conclusion to the arms race and the sniper's art.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Replaces the Particle Acceleration Rifle, or the Beam Marksman Rifle, with the Event Horizon Anti-Existential Beam Rifle.

In short, it shoots black holes.

It takes five cores to make.

It pretty much kills everything you point it at and their immediate bystanders as well.

You can do this from the relative safety of a whole four screens away thanks to it having a scope.

There can only ever be one vortex anomaly made in the round. The recipe, however, always requires the maximum number of cores that can be made in a round. If the number goes up for some reason (maybe someone wants to add more vortex items), so does the total number of cores needed to make an AEBR.

## Why It's Good For The Game

Firstly; the beam rifle? Awful. Jank code. Bad performance. Ancient balancing considerations. Just too strong to bring out regularly against specifically blobs, but completely fucking useless against anyone else.

So I'm hitting all these problems at once. I reduced a 600 line file to less than 60. I've removed all that bad code. I've also made it specifically unable to combat blobs at all because singularities send blobs off to safety.

Secondly; this is maybe the first example of 'more cores = bigger power'. The rifle was already utterly trash to have around, and I just didn't think the way it worked currently could really excite anyone enough to go on a big ol' quest to get it. Solution?

You will cause a shuttle call if you get this weapon.

## Changelog
:cl:
add: Replaces the Particle Acceleration Rifle with the Event Horizon anti-existential beam rifle. It shoots black holes. You can make this in-game. That's right, YOU!
balance: Only one vortex anomaly can be made in a round.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Don't merge this before https://github.com/tgstation/tgstation/pull/83892

**Comment by carlarctg** [comment]
> Only one vortex anomaly can be made in a round.

kinda dislike how this cucks the use of vortex anomalies for other things

what if it just turned the core into a random different core instead of refusing to make it

**Comment by necromanceranne** [comment]
Sounds like a good feature to me. You're welcome to make it, but this is actually the way the anomaly restrictions are meant to work.

**Comment by Astrogem2** [comment]
could we get a video to showcase this?

**Comment by Vect0r2** [comment]
This is actually so fucking cool I am quite honestly in shock
https://cdn.discordapp.com/attachments/705679956021215264/1250662515357253662/close_range.mp4?ex=666bc1bb&is=666a703b&hm=d9114e8560d160d2cbb70d949b8508faedd4f47894381b1bab5d756d4ce13ca0&

**Comment by CCC23** [comment]
Time to detonate NERDS with the FUNNY WEAPON

**Comment by Vect0r2** [comment]
> Time to detonate NERDS with the FUNNY WEAPON

if you are making this YOU are the nerd

File: code/modules/research/designs/weapon_designs.dm [hunk 1]
```diff
@@ -218,8 +218,8 @@
 	autolathe_exportable = FALSE
 
 /datum/design/beamrifle
-	name = "Beam Marksman Rifle Part Kit (Lethal)"
-	desc = "The gunkit for a powerful long ranged anti-material rifle that fires charged particle beams to obliterate targets."
+	name = "Event Horizon Anti-Existential Beam Rifle Part Kit (DOOMSDAY DEVICE)"
+	desc = "The kit that produces a weapon made to end your foes on an existsential level. Why the fuck can you make this?"
```
Comment (ATH1909) [inline code/modules/research/designs/weapon_designs.dm:222]: minor spelling mistake

i win

File: code/modules/projectiles/guns/energy/beam_rifle.dm [hunk 2]
```diff
@@ -1,588 +1,62 @@
...
-#define AIMING_BEAM_ANGLE_CHANGE_THRESHOLD 0.1
-
-/obj/item/gun/energy/beam_rifle
-	name = "particle acceleration rifle"
-	desc = "An energy-based anti material marksman rifle that uses highly charged particle beams moving at extreme velocities to decimate whatever is unfortunate enough to be targeted by one."
-	desc_controls = "Hold down left click while scoped to aim, when weapon is fully aimed (Tracer goes from red to green as it charges), release to fire. Moving while aiming or changing where you're pointing at while aiming will delay the aiming process depending on how much you changed."
+/obj/item/gun/energy/event_horizon
+	name = "\improper Event Horizon anti-existential beam rifle"
+	desc = "THe deranged minds of Nanotrasen, in their great hubris and spite, have birthed forth the definitive conclusion to the arms race. Weaponized black holes, and a platform to deliver them.\
+		To look upon this existential maleficence is to know that the pursuit of profit has consigned all life to this pathetic conclusion. The destruction of reality itself."
```
Comment (ATH1909) [inline code/modules/projectiles/guns/energy/beam_rifle.dm:4]:
```suggestion
		To look upon this existential maleficence is to know that the pursuit of profit has consigned all life to this pathetic conclusion: the destruction of reality itself."
```

**Comment by DaydreamIQ** [comment]
Would this replace any that happen to already be spawned in? I ask cause there's at least one bitrunner sim that has a particle rifle

**Comment by Vect0r2** [comment]
> 

it does

**Comment by necromanceranne** [comment]
> video?

vectOr posted one in the comments

**Comment by TagGamerGang22** [comment]
Something important to note is that cores can be ordered for free roundstart by scientists via their departmental order. If they're the RD then they get a ~6k departmental budget to spend roundstart which I believe will usually always be enough (or near enough) to order all 5 vortex cores.

**Comment by necromanceranne** [comment]
That's fine.

**Comment by necromanceranne** [comment]
Okay, more seriously; the design mandate for anomalies cores is currently 'if the amount of a given core is more limited, and an item needs more cores to work, then the item itself can become disproportionately stronger than normal gear as a result'.

It doesn't matter how easy it is, right now, to make five different cores and put it together to make the AEBR. That's a dial that can be adjusted as needed. But the end result can always be as overpowered as necessary, based on the amount of cores used, to justify the effort put in making all those cores and the potential for it to get swept out from under you by random chance or in-round circumstance.

The hurdles include;
1) Making the cores.
2) Getting the tech.
3) Getting the kit out of security's protolathe
4) Having enough minerals to print the kit.

If you don't want to encounter something that, potentially, you simply do not have an answer too while on the other side...well, in the spirit of absurdity, that's tough luck.

**Comment by Exester509** [comment]
This is absurd, stupidly powerful, a drain on station resources and extremely unwieldy to use without causing massive collateral damage.

Instant merge

**Comment by moocowswag** [comment]
Vortex armor is the best one so looks like this is never getting made.

**Comment by Sadboysuss** [comment]
> Vortex armor is the best one so looks like this is never getting made.

Pyro core does the same thing for reactive armor, L

**Comment by GPeckman** [comment]
> Would this replace any that happen to already be spawned in? I ask cause there's at least one bitrunner sim that has a particle rifle

Feature TBH

**Comment by necromanceranne** [comment]
No, all the ones mapped in have been replaced.

**Comment by CRITAWAKETS** [comment]
there is the issue that if you can refine one core usually you can refine 'em all, but whatever this thing looks really fun

**Comment by Singul0** [comment]
balancing this thing by limiting one of the core types to just one seems.... unwise for future PR's that will inevitably comes (core item PR's are really picking up steam after that omnibelt addition). barring an entire core type to just one functionality means the core gotta be super busted

**Comment by DATA-xPUNGED** [comment]
Soulful as fuck PR speedmerge, can't wait to never actually see this in practice

**Comment by moocowswag** [comment]
> Pyro core does the same thing for reactive armor, L

Thats placeholder

**Comment by Sadboysuss** [comment]
> > Pyro core does the same thing for reactive armor, L
> 
> Thats placeholder

vortex core making stealth armor is also a placeholder and doesn't make sense

**Comment by moocowswag** [comment]
> > > Pyro core does the same thing for reactive armor, L
> > 
> > 
> > Thats placeholder
> 
> vortex core making stealth armor is also a placeholder and doesn't make sense

Vortex always made stealth , pyro armor was op so core was made to make stealth

This pr will incentivize someone making yet another attempt at pyro armor being balanced since it hinges on void being valuable

**Comment by MrMelbert** [comment]
> Don't merge this before #83892

**Review by ZephyrTFA — APPROVED** [review]
pr of the year

**Comment by necromanceranne** [comment]
> balancing this thing by limiting one of the core types to just one seems.... unwise for future PR's that will inevitably comes (core item PR's are really picking up steam after that omnibelt addition).

Actually, I do have a solution for that. I could change the number of required vortexes to make a AEBR always equivalent to the maximum number of vortex cores.

**Comment by necromanceranne** [comment]
I did that

**Comment by necromanceranne** [comment]
> Vortex always made stealth , pyro armor was “op” so core was made to make stealth aswell
> 
> This pr will incentivize someone making yet another attempt at pyro armor being balanced since it hinges on void being valuable

Not for nothing, but maybe this should incentivize making stealth armor to compensate for the greater scarcity of the core.

**Comment by Loafin34** [comment]
This is the Event Horizon Anti-Existential Beam Rifle. It is a 150 pound, 5 foot long "rifle" that shoots an unfocused, undefinable mass of dark matter. It is a non-briggable item only because sec gave it a sporting exemption as a joke as if anybody is going to hunt with this. This thing would be overkill for hunting Nar'Sie.

I would like to paint a picture for you. It's 2AM and you hear a window break in your armory. This is the worst day this could happen, as every single one of your guns was lost in a tragic skateboarding accident this morning. All were lost except for one. You look across your room in dread at your anti-admin rifle. You know what you have to do, but you don't know if you have the strength to do it, both literally and figuratively.

Heaving the rifle into your arms, you charge the vortex core and begin to waddle towards the door. Your feet make a loud “thud” as you take each 6″ step. You know the spessmen hear you. You hope they do, for perhaps they will run and spare the world the suffering that is about to befall it.

You try to set the abomination down, but end up clipping your bedroom door and it is immediately knocked off its hinges by this battering ram in your hands. You attempt to round the corner, bonking the charging array against the airlock and adjacent wall across the hall at least 4 times. To your horror, a changeling and a traitor stand there at the end of the hall.

With a heavy heart, you raise the rifle to your shoulder while making inhuman grunting noises from the strain of attempting some semblance of a shooting position. The spessmen simply stare in disbelief, unable to process the situation they are witnessing, as if in a dream. You cannot physically aim it down sights, as the last time you fired it, it sucked your $3000 SecHUD GAR glasses in. You simply hold it at an angle that appears correct and fire.

You are immediately knocked to the floor as if hit by a chemist going 20 MPH. The shot connected with the ling and it erased him from existence. Even the memories of him have been destroyed and you're wondering why you just shot into an empty hallway. The shot continues to travel through space, taking at least half of perma, a solar array, and the cargo shuttle before being flung far away by a panicked admin, never to be seen again.

It is at this point, you realize you cannot hear. The surviving traitor can't hear either but he's also missing an arm from the blast and is currently vacating your brig. You don't care. Your shoulder is dislocated and there is a hole in your brand new secway You're crying now. The engiborgs arrive and, upon seeing the scene, start laughing. You start crying harder.

**Comment by necromanceranne** [comment]
>This thing would be overkill for hunting Nar'Sie.

Actually, you just reminded me that this gun can, in fact, kill Nar'sie.

Nice.

**Comment by Astrogem2** [comment]
The last thing a tider breaking into science after I spend an hour making this rifle:

Rifle-ezgif com-optimize

**Comment by Jacquerel** [comment]
Double check this to make sure it still does what you want it to do

**Comment by necromanceranne** [comment]
I'll have to make a new version of the black hole, so it might be best to put this on draft for now, I've got an important thing I have to prepare for this weekend.

**Comment by Admiral-Gigaton** [comment]
Balancing idea: take a page from paradise and make it take energy and ammo, said ammo being some bullshit like bluespace crystals or whatever, just so blac jhole can't rapid fire a fucking infinite mass object at Chang ling without running TGMC martini levels of skill

**Comment by DrAmazing343** [comment]
> Balancing idea: take a page from paradise and make it take energy and ammo, said ammo being some bullshit like bluespace crystals or whatever, just so blac jhole can't rapid fire a fucking infinite mass object at Chang ling without running TGMC martini levels of skill

I think it's perfect as it is. It's not meant to be balanced, it's meant to be a fucking world-ender predicated upon the fact that you spent like an hour tracking down the goddamn Dragon Balls to put together fucking Exodia

**Comment by Jacquerel** [comment]
Not sure what if anything you would want to do about this but it might be worth remembering that it is completely possible for 3/5 of the desired cores to have spawned at roundstart in science, you can (I think?) order the fourth basically immediately from the supply console, and so only need to buy one core. This still leaves you making 5 bombs but seems a lot less of a hurdle than you might expect from a 5 core requirement.

**Comment by necromanceranne** [comment]
The tech is fairly deep into the tree, and locked behind a few toxins experiments. To get there quickly and not get beaten to a pulp by the other scientists, you would have to actually engage in the toxins content itself as well.

This is also on top of mineral limits, and the major component needed, the kit, has to be printed out of security.

Not quite as simple as just the base five cores, otherwise, the rounds natural passive time gates will hopefully keep it in check.

File: code/datums/components/crafting/guncrafting.dm [hunk 3]
```diff
@@ -84,8 +84,10 @@
 	desc = "A suitcase containing the necessary gun parts to tranform a standard energy gun into a temperature gun. Fantastic at birthday parties and killing indigenious populations of lizardpeople."
 
 /obj/item/weaponcrafting/gunkit/beam_rifle
-	name = "particle acceleration rifle part kit (lethal)"
-	desc = "The coup de grace of guncrafting. This suitcase contains the highly experimental rig for a particle acceleration rifle. Requires an energy gun, a stabilized flux anomaly and a stabilized gravity anomaly."
+	name = "\improper Event Horizon anti-existnetial beam rifle part kit (DOOMSDAY DEVICE, DO NOT CONSTRUCT)"
```
Comment (Jacquerel) [inline code/datums/components/crafting/guncrafting.dm:87]:
```suggestion
	name = "\improper Event Horizon anti-existential beam rifle part kit (DOOMSDAY DEVICE, DO NOT CONSTRUCT)"
```

File: code/datums/components/crafting/guncrafting.dm [hunk 4]
```diff
@@ -84,8 +84,10 @@
...
+	desc = "What fevered minds wrought this terrible construction kit? To create a frame to harness the strange enemies that flow through the Spinward Sector towards such horrible acts of violence? \
```
Comment (Jacquerel) [inline code/datums/components/crafting/guncrafting.dm:88]:
```suggestion
	desc = "What fevered minds wrought this terrible construction kit? To create a frame to harness the strange energies that flow through the Spinward Sector towards such horrible acts of violence? \
```

File: code/datums/components/crafting/guncrafting.dm [hunk 5]
```diff
@@ -84,8 +84,10 @@
...
+		If you are unsound of mind, heed this. One vortex anomaly. One bluespace anomaly. Two flux anomalies. One gravity anomaly. And, maybe, if you believe in it, your soul itself. That is the price paid\
```
Comment (Jacquerel) [inline code/datums/components/crafting/guncrafting.dm:89]: May never come up but if we're making the potential cost in vortex cores variable, do we want to reference a specific number in the description?

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/datums/components/crafting/guncrafting.dm (hunk 3 above)
Comment (necromanceranne) [reply code/datums/components/crafting/guncrafting.dm:87]: Didn't I already fix this?

**Comment by adamkad1** [comment]
Cool, we got a new blob instakilling weapon

**Comment by optimumtact** [comment]
There will be no changes to anomaly cores/refined cores design so stop asking.

They are designed for a very specific purpose and all powerful weapons like this will be locked behind them.

it's not something that is up for debate unless headcoders (and them alone) want to do something with it.

**Comment by CCC23** [comment]
The throngler shall destroy all the healthes (security officers) in my holy domain (anywhere within a five mile radius)

**Pull request merged**